src/aiai_eval/task.py

```diff
@@ -54,7 +82,567 @@ def evaluate(self, model_id: str) -> Dict[str, dict]:
+                    min_max_length = min(list(all_max_lengths))
+                    tokenizer.model_max_length = min_max_length
+                else:
+                    tokenizer.model_max_length = 512
+            else:
+                tokenizer.model_max_length = 512
+
+        return dict(model=model, tokenizer=tokenizer)
+
+    def _load_spacy_model(self, model_config: ModelConfig) -> Dict[str, Any]:
+        """Load a spaCy model.
+
+        Args:
+            model_config (ModelConfig):
+                The configuration of the model.
+
+        Returns:
+            dict:
+                A dictionary containing at least the key 'model', with the value being
+                the model. Can contain other objects related to the model, such as its
+                tokenizer.
+        """
+        import spacy
+
+        # Ignore warnings from spaCy. This has to be called after the import, as the
+        # __init__.py file of spaCy sets the warning levels of spaCy warning W036
+        warnings.filterwarnings("ignore", module="spacy*")
+
+        local_model_id = model_config.model_id.split("/")[-1]
+
+        # Download the model if it has not already been so
+        try:
+            if not is_module_installed(local_model_id):
+                url = (
+                    f"https://huggingface.co/{model_config.model_id}/resolve/main/"
+                    f"{local_model_id}-any-py3-none-any.whl"
+                )
+                subprocess.run(["pip3", "install", url])
+
+        except CalledProcessError as e:
+            raise ModelFetchFailed(model_id=local_model_id, error_msg=e.output)
+
+        # Load the model
+        try:
+            model = spacy.load(local_model_id)
+        except OSError:
+            raise InvalidEvaluation(
+                f"The model {model_config.model_id} could not be installed from spaCy."
+            )
+        return dict(model=model)
+
+    def _adjust_label_ids(
+        self,
+        model: nn.Module,
+        model_config: ModelConfig,
+    ) -> nn.Module:
+        """Adjust the label ids of the model to match the dataset.
+
+        Args:
+            model (PyTorch Model):
+                The model to adjust the label ids of.
+            model_config (ModelConfig):
+                The model configuration.
+
+        Returns:
+            PyTorch Model:
+                The model with adjusted label ids.
+        """
+        # Placeholder for now.
+        return model
+
+    @abstractmethod
+    def _preprocess_data(self, dataset: Dataset, framework: str, **kwargs) -> Dataset:
+        """Preprocess a dataset by tokenizing and aligning the labels.
+        Args:
+            dataset (Hugging Face dataset):
+                The dataset to preprocess.
+            kwargs:
+                Extra keyword arguments containing objects used in preprocessing the
+                dataset.
+        Returns:
+            Hugging Face dataset: The preprocessed dataset.
+        """
```

Comment:
Also missing framework in docstring.

```suggestion
        """Preprocess a dataset by tokenizing and aligning the labels.
        
        Args:
            dataset (Hugging Face dataset):
                The dataset to preprocess.
            kwargs:
                Extra keyword arguments containing objects used in preprocessing the
                dataset.
                
        Returns:
            Hugging Face dataset: The preprocessed dataset.
        """
```